aries_cloudagent/admin/server.py

```diff
@@ -205,6 +207,17 @@ async def ready_middleware(request: web.BaseRequest, handler: Coroutine):
     raise web.HTTPServiceUnavailable(reason="Shutdown in progress")
 
 
+@web.middleware
+async def upgrade_middleware(request: web.BaseRequest, handler: Coroutine):
+    """Blocking middleware for upgrades."""
+    context: AdminRequestContext = request["context"]
+
+    if context._profile.name in upgrade_singleton.current_upgrades:
```

Comment:
The agent sets the singleton on startup based on the is_upgrading record in the wallet. So a new instance should have the correct values for this to work. At least that's what I was thinking.

Reply:
Are you talking instances of a given controller, or instances of ACA-Py?

Reply:
Instances of ACA-Py.

Reply:
I'm sure I can change the design slightly to make this work properly with multiple instances of acapy by using only the wallet(s) value instead of the singleton, but I really wanted to avoid that performance cost in the middleware.